custom_components/hubitat/fan.py

```diff
@@ -0,0 +1,76 @@
+"""Support for Hubitat fans."""
+
+from logging import getLogger
+import re
+from typing import Any, List, Optional
+
+from hubitatmaker import (
+    ATTR_SWITCH,
+    ATTR_SPEED,
+    CAP_FAN_CONTROL,
+    CMD_SET_LEVEL,
+    CMD_OFF, 
+    CMD_ON, 
+    CMD_CYCLE_SPEED, 
+    CMD_SET_SPEED,
+    DEFAULT_FAN_SPEEDS,
+    Device
+)
+
+from homeassistant.components.fan import (
+    FanEntity
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.util import color as color_util
+
+from .device import Hub, HubitatEntity, get_hub
+_LOGGER = getLogger(__name__)
+
+class HubitatFan(HubitatEntity, FanEntity):
+    """Representation of a Hubitat fan."""
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if the switch is on."""
+        return self.get_str_attr(ATTR_SWITCH) == "on"
+
+    @property
+    def speed(self) -> str:
+        """Return the speed of this fan."""
+        return self.get_str_attr(ATTR_SPEED)
+
+    @property
+    def speed_list(self) -> str:
+        """Return the list of speeds for this fan."""
+        return self._device.attributes[ATTR_SPEED].values or DEFAULT_FAN_SPEEDS
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
```

Comment:
```suggestion
    async def async_turn_on(self, speed: Optional[str] = None, **kwargs: Any) -> None:
```
The signature for `async_turn_on` in FanEntity has a named `speed` argument.

Reply:
Hmm... that should have been in there already.  Should be fixed now, though.